### ISSUE TITLE
Blinking timeline on loading

### ISSUE THREAD

**Issue by lachmanfrantisek** [nextcloud/social#572]
Sorry for creating another issue, but not sure if you know about this...

### Steps to reproduce
1. Install the current master.
2. Open the Global timeline.
3. Go down to see more messages.

### Expected behaviour

1. See the loading wheel.
2. See new messages after a while.

### Actual behaviour

- The timeline is blinking. (Also when scrolling up.)
- The new messages sometimes do not occur.
- The loading does not stop and takes a lot of resources.

<details><summary>Adding a lot of `t.item.actor_info is undefined` in browser log.</summary>

```
TypeError: "t.item.actor_info is undefined"
    TimelineContent https://my.nextcloud/apps2/social/js/4.bd941e0e52dd968e1b58.js:1
    _render https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    r https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    get https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    mn https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    $mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    $mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    init https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    h https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    Ko https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    _update https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    r https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    get https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    mn https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    $mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    $mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    init https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    b https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    O https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    O https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    Ko https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    _update https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    _update https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    r https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    get https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    run https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    hn https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    re https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    Zt https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    o https://my.nextcloud/core/js/dist/main.js?v=194e02da-1:230
    H https://my.nextcloud/core/js/dist/main.js?v=194e02da-1:230
    u https://my.nextcloud/core/js/dist/main.js?v=194e02da-1:235
social.js:7:11676
```

</details>

- See the gif bellow:

![social](https://user-images.githubusercontent.com/20214043/59084221-a7354b00-88fa-11e9-8a18-d832a2059b64.gif)

---

And one additional question. Is the `social` able to preview images? (Just to know if it is a bug or only the feature not implemented yet..;-)

## Server configuration detail

**Operating system:** Linux 4.4.0-143-generic #169-Ubuntu SMP Thu Feb 7 07:56:38 UTC 2019 x86_64

**Webserver:** nginx/1.15.12 (fpm-fcgi)

**Database:** pgsql PostgreSQL 10.8 on x86_64-pc-linux-musl, compiled by gcc (Alpine 8.3.0) 8.3.0, 64-bit

**PHP version:** 

7.3.5
Modules

 loaded: Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, gd, hash, iconv, json, mbstring, SPL, PDO, pdo_sqlite, bz2, posix, readline, Reflection, session, SimpleXML, standard, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, cgi-fcgi, apcu, Phar, exif, ftp, gmp, imap, intl, ldap, mysqli, pcntl, pdo_mysql, pdo_pgsql, pgsql, zip, imagick, redis, smbclient, libsmbclient, Zend OPcache

**Nextcloud version:** 16.0.1 - 16.0.1.1

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** hoellen/nextcloud docker image

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.2.0
 - activity: 2.9.1
 - admin_audit: 1.6.0
 - announcementcenter: 3.5.1
 - apporder: 0.7.1
 - audioplayer: 2.7.0
 - bruteforcesettings: 1.3.0
 - calendar: 1.7.0
 - camerarawpreviews: 0.7.0
 - carnet: 0.16.1
 - circles: 0.17.3
 - cloud_federation_api: 0.2.0
 - cms_pico: 0.9.8
 - comments: 1.6.0
 - contacts: 3.1.1
 - cospend: 0.1.0
 - data_request: 1.3.0
 - dav: 1.9.2
 - deck: 0.6.2
 - drawio: 0.9.3
 - drop_account: 0.1.0
 - dropit: 0.2.1
 - event_update_notification: 0.3.4
 - extract: 1.2.0
 - federatedfilesharing: 1.6.0
 - federation: 1.6.0
 - files: 1.11.0
 - files_3d: 0.1.0
 - files_downloadactivity: 1.5.0
 - files_external: 1.7.0
 - files_linkeditor: 1.0.10
 - files_markdown: 2.0.6
 - files_mindmap: 0.0.13
 - files_pdfviewer: 1.5.0
 - files_readmemd: 1.0.4
 - files_rightclick: 0.13.0
 - files_sharing: 1.8.0
 - files_texteditor: 2.8.0
 - files_trashbin: 1.6.0
 - files_versions: 1.9.0
 - files_videoplayer: 1.5.0
 - firstrunwizard: 2.5.0
 - forms: 1.0.3
 - gallery: 18.3.0
 - gpxedit: 0.0.11
 - gpxmotion: 0.0.9
 - gpxpod: 3.0.3
 - groupfolders: 4.0.2
 - guests: 1.0.0
 - impersonate: 1.3.0
 - issuetemplate: 0.5.0
 - logreader: 2.1.0
 - lookup_server_connector: 1.4.0
 - metadata: 0.9.0
 - music: 0.9.5
 - nextcloud_announcements: 1.5.0
 - notes: 2.6.0
 - notifications: 2.4.1
 - oauth2: 1.4.2
 - ownbackup: 19.5.0
 - password_policy: 1.6.0
 - passwords: 2019.6.0
 - polls: 0.10.2
 - privacy: 1.0.0
 - provisioning_api: 1.6.0
 - qownnotesapi: 19.4.0
 - quota_warning: 1.5.0
 - ransomware_detection: 0.5.1
 - ransomware_protection: 1.4.0
 - recommendations: 0.4.0
 - registration: 0.4.6
 - richdocuments: 3.3.8
 - serverinfo: 1.6.0
 - sharebymail: 1.6.0
 - sharerenamer: 2.6.0
 - social: 0.2.3
 - spreed: 6.0.1
 - survey_client: 1.4.0
 - suspicious_login: 1.0.0
 - systemtags: 1.6.0
 - tasks: 0.10.1
 - telephoneprovider: 1.0.2
 - theming: 1.7.0
 - twofactor_backupcodes: 1.5.0
 - twofactor_nextcloud_notification: 1.1.1
 - twofactor_totp: 2.1.2
 - twofactor_u2f: 3.0.0
 - updatenotification: 1.6.0
 - viewer: 1.0.0
 - workflowengine: 1.6.0
Disabled:
 - checksum
 - deck.old
 - encryption
 - end_to_end_encryption
 - phonetrack
 - support
 - twofactor_admin
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "apps_paths": [
        {
            "path": "\/nextcloud\/apps",
            "url": "\/apps",
            "writable": false
        },
        {
            "path": "\/apps2",
            "url": "\/apps2",
            "writable": true
        }
    ],
    "memcache.local": "\\OC\\Memcache\\APCu",
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "***REMOVED SENSITIVE VALUE***"
    ],
    "overwrite.cli.url":  "***REMOVED SENSITIVE VALUE***",
    "overwriteprotocol": "https",
    "dbtype": "pgsql",
    "version": "16.0.1.1",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "logtimezone": "Europe\/Prague",
    "logdateformat": "Y-m-d H:i:s",
    "mail_smtpmode": "smtp",
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpauth": 1,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "465",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpsecure": "ssl",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "maintenance": false,
    "theme": "",
    "loglevel": 0,
    "has_rebuilt_cache": true
}
```
</details>

**Are you using external storage, if yes which one:** 

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** 

## Client configuration

**Browser:** Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:67.0) Gecko/20100101 Firefox/67.0

**Operating system:** Fedora Workstation 30 (Gnome)

## Logs

<details><summary>Web server error log</summary>

```
...
```
</details>

<details><summary>Nextcloud log</summary>

```
...
```
</details>

<details><summary>Browser log</summary>

```
TypeError: "t.item.actor_info is undefined"
    TimelineContent https://my.nextcloud/apps2/social/js/4.bd941e0e52dd968e1b58.js:1
    _render https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    r https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    get https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    mn https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    $mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    $mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    init https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    h https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    Ko https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    _update https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    r https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    get https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    mn https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    $mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    $mount https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    init https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    d https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    b https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    O https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    O https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    Ko https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    _update https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    _update https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    r https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    get https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    run https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    hn https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    re https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    Zt https://my.nextcloud/apps2/social/js/social.js?v=194e02da-1:7
    o https://my.nextcloud/core/js/dist/main.js?v=194e02da-1:230
    H https://my.nextcloud/core/js/dist/main.js?v=194e02da-1:230
    u https://my.nextcloud/core/js/dist/main.js?v=194e02da-1:235
social.js:7:11676
```
</details>

**Comment by daita** [comment]
@juliushaertl  any idea ?

**Comment by juliushaertl** [comment]
Looks like it is caused by the transition that is used to show new posts. It should probably be limited to only happen when creating own posts.

**Comment by lachmanfrantisek** [comment]
I found another problem triggered by this:

The messages from the global timeline are shown also in the local timeline. It should be empty in the following example.

Steps to reproduce:
1. select local timeline (no messages ✔)
2. select global timeline (some messages ✔)
3. select local timeline (no messages ✔)
4. select global timeline (some messages ✔)
5. scroll down (blinking and no new message)
6. select local timeline (messages from the global timeline are shown ❌)

![social2](https://user-images.githubusercontent.com/20214043/59536252-799c6300-8ef3-11e9-8e48-fac041112991.gif)

**Comment by daita** [comment]
do you still have blinking issue if you unfollow Federation Bot ?

**Comment by lachmanfrantisek** [comment]
@daita I need to fix the migration problems first to check it properly. (I cannot easily edit the follows...)

Home timeline without FederationBot works well, but the global timeline with FederationBot has the loading problem...

**Comment by daita** [comment]
Can you give me that full social address of the Federation Bot ? is it an account I can follow myself ?

**Comment by lachmanfrantisek** [comment]
Of course:
@federationbot@mastodon.host
https://mastodon.host/@federationbot

It produces a lot of spam/messages...;)

I've been following it during my setup because it is very active...

**Comment by daita** [comment]
I am rewriting the Announce stream (#586) and it should fix this issue. I will do some test on big scale on my side to confirm the improvement.

**Comment by lachmanfrantisek** [comment]
> I am rewriting the Announce stream (#586) and it should fix this issue. I will do some test on big scale on my side to confirm the improvement.

Thank you!

**Comment by lachmanfrantisek** [comment]
@daita 

Thanks for the update. I've tried to upgrade with the following result:

1. Update in the web console failed.
2. Update in the console took a long time. (Maybe a reason for the failure in the browser..?)
3. Getting a following error in the browser:

```
Failed to load more timeline entries Error: "Request failed with status code 500"
    exports https://my.nextcloud/apps2/social/js/social.js?v=11a7e9aa-9:153
    exports https://my.nextcloud/apps2/social/js/social.js?v=11a7e9aa-9:160
    onreadystatechange https://my.nextcloud/apps2/social/js/social.js?v=11a7e9aa-9:153
4.bd941e0e52dd968e1b58.js:1:31362
```

(Seeing this in all timelines and no messages are loaded..)

**Comment by daita** [comment]
ok, let's see how the timeline handle within the next hours, after more Post are generated. The issue could comes from old posts.

**Comment by lachmanfrantisek** [comment]
OK, I'll let you know...

**Comment by daita** [comment]
Anyway, if I were you, after a backup of the table `oc_social_a2_stream` I would try to run:

`DELETE FROM oc_social_a2_stream WHERE type='Announce';`

**Comment by lachmanfrantisek** [comment]
> Anyway, if I were you, after a backup of the table oc_social_a2_stream I would try to run:

Thanks for the tip -- I'll try that tomorrow and let you know.

> ok, let's see how the timeline handle within the next hours, after more Post are generated. The issue could comes from old posts.

As by now, still no new messages in the timeline....

**Comment by daita** [comment]
If, after running the SQL request, you still can't see anything, you should empty the table totally (keep a backup, in case), and see if it fixes your issue. Emptying the table means that old posts will be lost, but new posts should be available after this.

**Comment by daita** [comment]
Please confirm issue is fixed in 0.2.5

**Comment by lachmanfrantisek** [comment]
Sorry for my delay. I had a very busy week.

In `0.2.6`, the blinking and loading problems disappeared and I'm seeing new posts. (I do not need to remove anything.) Thank you for the fix!

The weird thing is, that the loading stops working when I see the message by Federationbot (8 days old). Maybe, it will work correctly in the future. If you think it's not relevant anymore, we can close this issue...
